ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -253,8 +242,8 @@ forkSyncStateOnTipPointChange registry menv =
 --
 -- See the necessary invariants on the Haddock for 'API.tryAddTxs'.
 --
--- This function does not sync the Mempool contents with the ledger state in
--- case the latter changes, it relies on the background thread to do that.
+-- This function syncs the Mempool contents with the ledger state in case the
+-- ledger state in the Mempool is no longer accessible for the LedgerDB.
```

Comment:
I'm not sure I follow. From the point of the mempool the only important thing is that we cannot access that ledger state, that's all. Otherwise we get into explaining why those values are not accessible etc.

Reply:
Let's expand this comment explaining why we should resync if we have a stale ledger state (we would resync in any case, but we ensure it happens before retrying) (also we must always apply the transaction on some ledger state even if it gets rejected or accepted, so to keep this behavior we must resync)